clang/utils/TableGen/ClangAttrEmitter.cpp

```diff
@@ -189,7 +189,7 @@ static StringRef NormalizeGNUAttrSpelling(StringRef AttrSpelling) {
 
 typedef std::vector<std::pair<std::string, const Record *>> ParsedAttrMap;
 
-static ParsedAttrMap getParsedAttrList(const RecordKeeper &Records,
```

Comment:
I expected to see more `const` instead of less. Why is this here (and below) passes as non-const?

Reply:
Please see the discourse thread: https://discourse.llvm.org/t/changing-tablegen-getallderiveddefinitions-to-return-arrayref-const-record/80586/5

Changing all backends to use const in a single go will be too large a change, so this is a way to stage it. Over time, all backends should move to const variant, and then the non-const ones can be deleted.

Reply:
Would be good to have those conversions queued up to see how this progresses towards that goal

Reply:
I will work on some of them, but not sure if I have the bandwidth to address all of them (Maybe the first few might be more involved and then we will have a better idea of the scope). These will be in similar vein to what I have already done for the IntrinsicEmitter and JSON/Detailed record emitters.

Reply:
I propose the following next steps (once this is committed):

1. We need to send a PSA as current downstream backends may break due to this (fix is easy)
2. I'll work on migrating on of each MLIR. Clang, and may be another LLVM backends to use const.
3. For the rest, we need to chip away. May be we try to recruit volunteers?
4. Before we can deprecate the non-const functions completely, give enough time for downstream backends to change. Maybe this should be a part of the PSA in (1).
5. Delete the non-const functions here, and any other const vs non-const difference we may have introduced during the transition.

Reply:
Here's an example of migrating one MLIR backend: https://github.com/llvm/llvm-project/commit/8ffa5e92662db6aa66a7520da80a360856ac1440

Reply:
And here's for the entire MLIR tablegen code: https://github.com/llvm/llvm-project/commit/061076db7f668f67183e813db6d339a840a05414

So may be its not that involved. Note that this does not address all the const correctness issues, as const Record* function may themselves non-const pointers. So that needs to be fixed gradually as well. So I propose we get this commit in, and then the MLIR one as a follow on. I will see if I can do the clang one as well.

Reply:
And for all clang-tablegen its here: https://github.com/llvm/llvm-project/commit/abf19e5d26ccbf0714fbb93dfc8c783da9e407a1

So looks like we may be able to migrate quickly, may be wait for a few days (weeks?) for downstream adoption and delete the non-const overloads.